test/test_helper.rb

```diff
@@ -56,6 +56,7 @@ def before_setup
 
   def after_teardown
     Resque::Worker.kill_all_heartbeat_threads
+    sleep 0.1 until Thread.list.count == 1
```

Comment:
we might need to worry about a similar leak in production.  We should join on the threads to make sure they die. That is also needed to properly unregister the worker and cleanup its heartbeat